Sources/Bot/API/Resource.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+import Result
+
+internal struct Resource<T> {
+    typealias Decoder = (Response) -> Result<T, GitHubClient.Error>
+
+    internal var method: Method
+    internal var path: String
+    internal var queryItems: [URLQueryItem]
+    internal var headers: [String: String]
+    internal var body: Data?
+    internal let decoder: Decoder
```

Comment:
This was necessary to support the merge of branches which can finish with three valid codes:

- 200: Merged
- 204: Nothing to merge
- 409: Conflict

I tried to merge `200` and `204` in a resource that would return an optional `Commit` but that doesn't work because with a `204` we don't receive a body leading to a failure during decoding.